packages/react-server/core/ClientController.js

```diff
@@ -151,10 +156,76 @@ class ClientController extends EventEmitter {
 				this.context.registerRequestLocal();
 			}
 		}
+
+		// If this is a History.events.PUSHSTATE navigation,
+		// and we have control of the navigation bar (we're
+		// not in a frameback frame) we should change the URL
+		// in the bar location bar before rendering.
```

Comment:
Heh, wonder how long [that's been there](https://github.com/redfin/react-server/pull/97/files#diff-db9055542b789c676642bfe7f729f1ecL175).  Will fix.